DataFormats/PatCandidates/src/CovarianceParameterization.cc

```diff
@@ -0,0 +1,225 @@
+ fileToRead_ = TFile::Open(fip.fullPath().c_str()); 
+ TFile & fileToRead=*fileToRead_;
+//Read files from here fip.fullPath().c_str();
+ if(fileToRead.IsOpen())  {
+     readFile(fileToRead);
+
+     TIter next(((TDirectoryFile*)fileToRead.Get("schemas"))->GetListOfKeys());
+     TKey *key;
+     while ((key = (TKey*)next())) {
+      TClass *cl = gROOT->GetClass(key->GetClassName());
+      if (!cl->InheritsFrom("TDirectoryFile")) continue;
+      std::string schemaNumber = key->ReadObj()->GetName();
+      uint16_t schemaN = std::stoi(schemaNumber);
+     //for (int folderNumber = 0; folderNumber < 6 ; folderNumber++) {
+      CompressionSchema schema; 
+      for (int i = 0; i < 5; i++) {
+            for (int j = i; j < 5; j++) {        //FILLING ONLY THE SCHEMA OF SOME ELEMENTS
+                  std::string folder = "schemas/" + schemaNumber + "/"  + char(48+i) + char(48+j);
+//                fileToRead.Get(folder.c_str())->ls();
+                std::string methodString = folder + "/method";
+                std::string targetString = folder + "/target";
+                std::string bitString = folder + "/bit";
+                std::vector<float> vParams ;
+                TVector *p=(TVector*) fileToRead.Get((folder+"/param").c_str());
+                for(int k = 0 ; k < p->GetNoElements() ; k++){
+                    vParams.push_back((*p)[k]);
+                }
+
+                schema(i,j)=CompressionElement((CompressionElement::Method) ((TParameter<int>*) fileToRead.Get(methodString.c_str()))->GetVal(),(CompressionElement::Target) ((TParameter<int>*) fileToRead.Get(targetString.c_str()))->GetVal(), (int) ((TParameter<int>*) fileToRead.Get(bitString.c_str()))->GetVal(), vParams);
+        
+                
+            }
+        }
+     schemas[schemaN]=schema; 
+     }
+//      schemas.push_back(schema0);
+
+/* //Override miniaod schema as in root file
+     CompressionSchema schemaMiniAOD;
+     schemaMiniAOD(0,0)=CompressionElement(CompressionElement::logPack,CompressionElement::ratioToRef,256,{-5,5});
+     schemaMiniAOD(1,1)=schemaMiniAOD(0,0);
+     schemaMiniAOD(2,2)=schemaMiniAOD(0,0);
+     schemaMiniAOD(3,3)=CompressionElement(CompressionElement::float16,CompressionElement::realValue,16,{10000});
+     schemaMiniAOD(4,4)=CompressionElement(CompressionElement::float16,CompressionElement::realValue,16,{10000});
+     schemaMiniAOD(3,4)=CompressionElement(CompressionElement::float16,CompressionElement::realValue,16,{10000}); 
+     schemaMiniAOD(2,3)=schemaMiniAOD(0,0);
+     schemaMiniAOD(1,4)=schemaMiniAOD(0,0);
+
+     schemas[0]=(schemaMiniAOD); */
+
+
+    loadedVersion_=version; 
+ } else {loadedVersion_=-1;}
+
+}
+
+
+
+void CovarianceParameterization::readFile( TFile & f) {
+
+    for (int i = 0; i < 5; i++) {
+        for (int j = i; j < 5; j++) {
+
+            std::string String_first_positive = "_pixel_";
+            std::string String_second_positive = "_noPixel_";
```

Comment:
I think this part of the code should just be rewritten a bit better (but it is called once per cmsRun instance so it is not top of my priorities)